src/hdusd/bl_nodes/node_parser.py

```diff
@@ -65,6 +65,11 @@ def set_input(self, name, value):
 
         val_data = value.data if isinstance(value, NodeItem) else value
         nd_input = self.nodedef.getInput(name)
+
+        if not isinstance(val_data, NodeItem) and not isinstance(val_data, mx.Node) \
+                and nd_input.getType() != type(val_data).__name__ and not nd_input.getType() == 'filename':
+            return
+
```

Comment:
We should set value instead of ignoring it. Add type check to mx_utils.set_param_value